### ISSUE TITLE
Hidden struct copy analyzer not always accurate

### ISSUE THREAD

**Issue by Thealexbarney** [SergeyTeplyakov/ErrorProne.NET#158]
These sample programs don't produce the correct result inside `Program.GetValue()`

Incorrect EPS06 warning
```cs
namespace Test
{
    public struct TestStruct
    {
        public int Value { get; set; }
        private long _value2;
    }

    public static class Program
    {
        public static int GetValue(in TestStruct test)
        {
            return test.Value;
        }
    }
}
```
Incorrect EPS06 warning
```cs
namespace Test
{
    public struct TestStruct
    {
        private int _value;
        private long _value2;

        public readonly int GetValue() => _value;
    }

    public static class Program
    {
        public static int GetValue(in TestStruct test)
        {
            return test.GetValue();
        }
    }
}
```
The struct has a hidden copy but no EPS06 warning
```cs
namespace Test
{
    public struct TestStruct
    {
        private int _value;

        public int GetValue() => _value;
    }

    public static class Program
    {
        public static int GetValue(in TestStruct test)
        {
            return test.GetValue();
        }
    }
}
```

**Comment by SergeyTeplyakov** [comment]
The analyzers were implemented before readonly members were added to the language.

This will be resolved in the next release.

**Comment by SergeyTeplyakov** [comment]
Fixed by #187.

Please note, that the following code does not emit the diagnostic:
```csharp
namespace Test
{
    public struct TestStruct
    {
        private int _value;

        public int GetValue() => _value;
    }

    public static class Program
    {
        public static int GetValue(in TestStruct test)
        {
            return test.GetValue();
        }
    }
}
```

Because `TestStruct` is too small. I added a threshold (that now is configurable from .editorconfig, thanks to #185) but by default, the diagnostics are emitted only for structs larger 3 * sizeof(long) to avoid too many diagnostics for structs that won't cause any real performance issues.
